integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+                    .select(document.name.as("blep"), document.name.substring(0, 2))
+                    .where(document.name.length().gt(param))
+                    .set(param, 1);
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    public void testParameterExpressionInSelect() {
+        doInJPA(entityManager -> {
+            Param<Integer> param = new Param<>(Integer.class, "theSuperName");
+
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("blep"), param)
+                    .where(document.name.length().gt(param))
+                    .set(param, 1);
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    public void testSubQuery() {
+        doInJPA(entityManager -> {
+            QDocument sub = new QDocument("sub");
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("blep"), document.name.substring(0, 2))
+                    .where(document.id.in(select(sub.id).from(sub)));
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    @Category({ NoMySQLOld.class })
+    public void testWindowFunction() {
+        Assume.assumeTrue(dbmsDialect.supportsWindowFunctions());
+
+        doInJPA(entityManager -> {
+            QDocument sub = new QDocument("sub");
+
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .select(document.name.as("documentName"), rowNumber().over().orderBy(document.id), lastValue(document.name).over().partitionBy(document.id).orderBy(document.id))
+                    .where(document.id.in(select(sub.id).from(sub)));
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    @Category({ NoMySQLOld.class })
+    public void testNamedWindowFunction() {
+        Assume.assumeTrue(dbmsDialect.supportsWindowFunctions());
+
+        doInJPA(entityManager -> {
+            QDocument sub = new QDocument("sub");
+
+            NamedWindow namedWindow = new NamedWindow("namedWindow").partitionBy(document.id).orderBy(document.id);
+
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .window(namedWindow)
+                    .select(document.name.as("documentName"), rowNumber().over(namedWindow), lastValue(document.name).over(namedWindow))
+                    .where(document.id.in(select(sub.id).from(sub)));
+
+            List<Tuple> fetch = query.fetch();
+            assertFalse(fetch.isEmpty());
+        });
+    }
+
+    @Test
+    @Category({NoMSSQL.class})
```

Comment:
Please add a comment why it doesn't work. I assume your limit without ordering is the issue. You should always define a order by clause when using limit.